libs/sdk-core/src/lsp.rs

```diff
@@ -73,31 +75,29 @@ impl LspInformation {
         Ok(info)
     }
 
-    /// Returns the channel opening fees, either the ones provided by the user (if any), or the ones from LSP.
+    /// Returns the cheapeset opening channel fees from LSP that within the expiry range.
     ///
     /// If the LSP fees are needed, the LSP is expected to have at least one dynamic fee entry in its menu,
     /// otherwise this will result in an error.
-    pub(crate) fn choose_channel_opening_fees(
-        &self,
-        maybe_user_supplied_fee_params: Option<OpeningFeeParams>,
-        fee_type: DynamicFeeType,
-    ) -> SdkResult<OpeningFeeParams> {
-        match maybe_user_supplied_fee_params {
-            // Validate given opening_fee_params and use it if possible
-            Some(user_supplied_fees) => {
-                user_supplied_fees.validate()?;
-                Ok(user_supplied_fees)
-            }
-            // We pick our own if None is supplied
-            None => match fee_type {
-                DynamicFeeType::Cheapest => self
-                    .opening_fee_params_list
-                    .get_cheapest_opening_fee_params(),
-                DynamicFeeType::Longest => {
-                    self.opening_fee_params_list.get_48h_opening_fee_params()
+    pub(crate) fn cheapest_open_channel_fee(&self, expiry: u64) -> SdkResult<&OpeningFeeParams> {
+        for fee in &self.opening_fee_params_list.values {
+            match fee.valid_until_date() {
+                Ok(valid_until) => {
+                    if valid_until > Utc::now().add(Duration::seconds(expiry as i64)) {
+                        return Ok(fee);
+                    }
                 }
-            },
+                Err(e) => {
+                    return Err(SdkError::CalculateOpenChannelFeesFailed { err: e.to_string() })
+                }
+            }
         }
+        self.opening_fee_params_list
+            .values
+            .last()
```

Comment:
That's a good point, I started from error and then ended up with this.
If we play strict here and return error then we are totally dependent on the lsp expiration coverage.
So if for example the lsp max validity is 48h and we ask for 3 days in the sdk we won't be able to create invoice.
Also users won't be able to create invoices with 3 days expiration for example.
While strictly speaking that might be correct, the experience being less optimal and the developer would need to query for the fees and deduce from that the final range so he/she can limit the expiration in their software.

The other alternative is to define the behavior as "finding the most suitable fee for that query" and if we don't have something that match exactly we will return the one that has the highest chances to success. The fact that we return the fee params in the result let the user observe the gap so we don't hide anything. 
A bit more pragmatic approach towards the edge cases but keeps the experience smooth.
What do you think @JssDWt ?

Reply:
I think an invoice with an expiry of 3 days doesn't make sense if the fee params are valid for two days, because you won't be able to pay the invoice after two days anyway if the fee params expire. If this happens, we should ask the LSP to include fee params that are valid longer, and otherwise use a different LSP.

Reply:
Done